algorithmic_efficiency/workloads/ogbg/ogbg_jax/workload.py

```diff
@@ -96,9 +96,10 @@ def _binary_cross_entropy_with_mask(
     return jnp.where(mask, losses, 0.)
 
   def _eval_metric(self, labels, logits, masks):
-    loss, _ = self.loss_fn(labels, logits, masks)
+    loss = self.loss_fn(labels, logits, masks)
+    mean_loss = loss['summed'] / loss['n_valid_examples']
```

Comment:
Ok, makes sense. But I'm not sure if we may also have batch level padding during training? Since there will be no remainder there should not be padding graphs, but maybe only padding nodes and edges?

Reply:
`jraph.dynamically_batch` calls `pad_with_graphs` [here](https://github.com/deepmind/jraph/blob/master/jraph/_src/utils.py#L1005) which could add some padding during training if we go over the max number of nodes or edges, right? I may be misreading the `jraph` lib code though.

Reply:
I think you are right. So there seem to be two different issues with ogbg:
1. Eval/train losses and grads are not the same for Jax and PyTorch in the traindiffs test.
3. The evaluation loss is not averaged correctly (your comment here; for training the loss average should be correct due to this PR).